scripts/update-index-alias.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+"""
+A script to update the search index an alias is assoiciated with.
+
+To delete old indexes (those with no alias), set --delete-old-indexes=yes
+
+Usage:
+scripts/update-index-alias.py <alias> <target> <stage> <search-api-endpoint> [options]
+
+Options:
+    --delete-old-indexes=<delete-old-indexes>  [default: no]
+"""
+import os
+import sys
+import subprocess
+import yaml
+import json
+import requests
+from requests.exceptions import HTTPError
+from docopt import docopt
+
+
+def update_index_alias(alias, target, stage, endpoint, delete_old_indexes):
+    auth_token = _get_auth_token(stage)
+    headers = {
+        'Authorization': "Bearer {}".format(auth_token),
+        'Content-type': 'application/json'
+    }
+
+    _rename_current_alias_to_old(alias, endpoint, headers)
+
+    url = "https://{}/{}".format(endpoint, alias)
+    data = {'type': 'alias', 'target': "{}".format(target)}
+
+    response = requests.put(url, headers=headers, json=data)
+    _check_response_status(response, 'updating alias')
+
+    if delete_old_indexes == 'yes':
+        _delete_old_indexes(endpoint, auth_token)
+
+
+def _get_auth_token(stage):
+    DM_CREDENTIALS_REPO = os.environ.get('DM_CREDENTIALS_REPO')
+    creds = subprocess.check_output([
+        "{}/sops-wrapper".format(DM_CREDENTIALS_REPO),
+        "-d",
+        "{}/vars/{}.yaml".format(DM_CREDENTIALS_REPO, stage)
+    ])
+    auth_tokens = yaml.load(creds)['search_api']['auth_tokens']
+    dev_token = [token for token in auth_tokens if token[0] == 'D']
+    return dev_token[0]
+
+
+def _rename_current_alias_to_old(alias, endpoint, headers):
+    old_alias = "old-{}".format(alias)
```

Comment:
I'd probably prefer to have it called `"{}-old"` so that it appears in the same place as the current index. Only really matters when we start thinking about having lots more frameworks/catalogues indexed, but I think in that situation they'd make more sense to keep together.